extensions/amp-analytics/0.1/visibility-impl.js

```diff
@@ -531,10 +536,20 @@ export class Visibility {
   }
 }
 
+
 /**
- * @param  {!Window} win
+ * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
  * @return {!Visibility}
  */
-export function installVisibilityService(win) {
-  return fromClass(win, 'visibility', Visibility);
+function installVisibilityServiceForDoc(ampdoc) {
+  return fromClassForDoc(ampdoc, 'visibility', Visibility);
 };
+
+
+// Register doc-service factory.
+AMP.registerServiceForDoc(
```

Comment:
You mean dropping registering visibility service all together? Yeah it seems its only used in instrumentation so we can drop it. Is that what you meant?